anti-tracking-threat-model.md

```diff
@@ -0,0 +1,196 @@
+   `adtech.example` cookie holding a unique ID, even in browsers with
+   multi-keyed caches.
+1. AdTech can convince News to give them a `news.example` certificate's private
+   key or to tell a CA that AdTech has permission to receive certificates valid
+   to sign exchanges for `news.example`.
+
+   This seems like an implausible capability at first glance, but publishers
+   routinely give CDNs this ability to terminate TLS traffic, and a
+   generally-trusted AdTech could convince publishers that it's merely doing
+   what a good CDN would, for cheaper.
+1. `adtech.example` can serve a link to `news.example` and expect users to click
+   on it and then browse around `news.example`.
+   1. This link can point to a resource that redirects to `news.example`.
+   1. This link can point to a signed exchange or web package containing
+      content signed by `news.example`'s certificate.
+1. AdTech can make any number of signatures with private keys it controls.
+
+## Attacker Non-capabilities
+
+1. AdTech cannot manipulate the DNS system on a per-user basis.
+1. There is a limit to the complexity AdTech can convince News to add to its
+   serving infrastructure. For example, News is willing to ignore unknown query
+   parameters and fragments, but is not willing to ignore unexpected path
+   segments. TODO: Can we describe this limit any more precisely?
+
+## Attacker goals that we want to frustrate
+
+1. The user does not want AdTech to be able to augment its profile of them while
+   reading articles on `news.example`.
+1. The user does not want AdTech's rich profile of them to influence the content
+   of ads or articles on `news.example`.
+1. More abstractly, AdTech cannot transfer its unique ID for a user to
+   the Javascript environment created for `news.example`.
+   1. Failing this, the user agent or external auditors should be able to detect
+      that AdTech is tracking users.
+
+# Attacks and their Mitigations
+
+This section lists potential ways AdTech might achieve its unwanted goals, along
+with proposed or already-adopted mitigations to frustrate those goals. Not all
+of the attacks use Web Packaging, so that we can explore the threat model's
+implications for the rest of the web platform.
+
+## Sign identifying information into the package
+
+### The Attack
+
+1. AdTech convinces News to give AdTech a package-signing certificate for
+   `news.example`, perhaps by offering to handle the technical complications of
+   signing packages.
+   1. Alternately, News could set up a service that signs packages AdTech sends
+      it.
+1. When a user clicks a link from `adtech.example` to
+   `https://adtech.example/news.example.sxg`, they send identifying information
+   to the `adtech.example` server. This could be cookies or a user ID encoded in
+   the query, path, or even hostname.
+1. Instead of signing `news.example`'s content directly, AdTech embeds the
+   user's identity in that content and signs the result on the fly.
```

Comment:
With Server Timing, we have response headers that are directly readable from JS, so embedding the UID through them might be feasible even without on-the-fly signing. So we need to make sure that the distributor cannot add arbitrary non-signed headers to the internal response, as well as that the external response Server Timing headers are not exposed to the navigated page.
I believe that's already the case, tbh, but IMO it's worthwhile to explicitly note that.

Reply:
It's a security requirement that the distributor can't add response headers to the post-redirect response. https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#signed-headers mentions that idea, although it's in a section nobody's implementing. The application/signed-exchange format just can't represent such unsigned post-redirect response headers.

I don't think Server-Timing response headers on a 303 response get exposed to the post-redirect javascript. So I'm not sure there's really anything to say here.